crates/ruff_linter/src/rules/flake8_pie/rules/unnecessary_placeholder.rs

```diff
@@ -83,6 +83,12 @@ pub(crate) fn unnecessary_placeholder(checker: &mut Checker, body: &[Stmt]) {
         return;
     }
 
+    // In a type-checking block, a trailing ellipsis might be meaningful. A
+    // user might be using the type-checking context to declare a stub.
+    if checker.semantic().in_type_checking_block() {
+        return;
+    }
```

Comment:
Should this be down at the `in_protocol_or_abstract_method` line? (I.e., shouldn't this only apply to ellipses, not stubs?)